app/feature/feature-profile/src/main/kotlin/com/hedvig/android/feature/profile/aboutapp/AboutAppViewModel.kt

```diff
@@ -1,41 +1,77 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import com.apollographql.apollo.ApolloClient
 import com.hedvig.android.apollo.ErrorMessage
 import com.hedvig.android.apollo.safeFlow
 import com.hedvig.android.core.datastore.DeviceIdDataStore
-import kotlin.time.Duration.Companion.seconds
-import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.WhileSubscribed
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
 import octopus.MemberIdQuery
 
 internal class AboutAppViewModel(
   apolloClient: ApolloClient,
   deviceIdDataStore: DeviceIdDataStore,
-) : ViewModel() {
-  val uiState: StateFlow<AboutAppUiState> = combine(
-    apolloClient
-      .query(MemberIdQuery())
-      .safeFlow(::ErrorMessage)
-      .map { it.getOrNull()?.currentMember?.id },
-    deviceIdDataStore.observeDeviceId(),
-  ) { memberId, deviceId ->
-    AboutAppUiState(memberId, deviceId)
+) : MoleculeViewModel<AboutAppEvent, AboutAppUiState>(
+  initialState = AboutAppUiState.Loading,
+  presenter = AboutAppPresenter(apolloClient, deviceIdDataStore),
+)
+
+private class AboutAppPresenter(
+  private val apolloClient: ApolloClient,
+  private val deviceIdDataStore: DeviceIdDataStore,
+) : MoleculePresenter<AboutAppEvent, AboutAppUiState> {
+  @Composable
+  override fun MoleculePresenterScope<AboutAppEvent>.present(lastState: AboutAppUiState): AboutAppUiState {
+    var currentState by remember { mutableStateOf(lastState) }
+    var loadIteration by remember { mutableIntStateOf(0) }
+
+    CollectEvents { event ->
+      when (event) {
+        AboutAppEvent.Retry -> loadIteration++
+      }
+    }
+
+    LaunchedEffect(loadIteration) {
+      currentState = AboutAppUiState.Loading
+      combine(
+        apolloClient
+          .query(MemberIdQuery())
+          .safeFlow(::ErrorMessage)
+          .map { it.getOrNull()?.currentMember?.id },
+        deviceIdDataStore.observeDeviceId(),
+      ) { memberId, deviceId ->
+        AboutAppUiState.Content(memberId, deviceId)
+      }.collectLatest { state ->
+        currentState = state
+      }
+    }
+
+    return currentState
   }
-    .stateIn(
-      viewModelScope,
-      SharingStarted.WhileSubscribed(5.seconds),
-      AboutAppUiState(null, null),
-    )
 }
 
-internal data class AboutAppUiState(
-  val memberId: String?,
-  val deviceId: String?,
-)
+internal sealed interface AboutAppEvent {
+  data object Retry : AboutAppEvent
```

Comment:
If this is never used at all we could just pass `Unit` instead of `AboutAppEvent` in the presenter/VM and not call `CollectEvents` at all